File: VB/dxSample/Custom Layout Control/CustomLayoutControlImplementor.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraLayout
Imports DevExpress.XtraLayout.Registrator

Namespace dxSample
    Friend Class CustomLayoutControlImplementor
        Inherits LayoutControlImplementor

        Public Sub New(ByVal owner As ILayoutControlOwner)
            MyBase.New(owner)
        End Sub

        Public Overrides Function CreateLayoutItem(ByVal parent As LayoutGroup) As BaseLayoutItem
            Return New CustomLayoutControlItem(CType(parent, LayoutControlGroup))
        End Function
        Protected Overrides Sub InitializePaintStyles()
            Dim lookAndFeelOwner As ISupportLookAndFeel = owner.GetISupportLookAndFeel()
            If lookAndFeelOwner IsNot Nothing Then
                PaintStyles.Add(New LayoutOffice2003PaintStyle(lookAndFeelOwner))
                PaintStyles.Add(New LayoutWindowsXPPaintStyle(lookAndFeelOwner))
                PaintStyles.Add(New CustomLayoutSkinPaintStyle(lookAndFeelOwner))
                PaintStyles.Add(New Style3DPaintStyle(lookAndFeelOwner))
                PaintStyles.Add(New UltraFlatPaintStyle(lookAndFeelOwner))
                PaintStyles.Add(New FlatPaintStyle(lookAndFeelOwner))
            End If
            lookAndFeelOwner = Nothing
        End Sub
    End Class
End Namespace

File: CS/dxSample/Custom Layout Control/CustomLayoutControlImplementor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.LookAndFeel;
using DevExpress.XtraLayout;
using DevExpress.XtraLayout.Registrator;

namespace dxSample {
    internal class CustomLayoutControlImplementor : LayoutControlImplementor {
        public CustomLayoutControlImplementor(ILayoutControlOwner owner)
            : base(owner) {
        }

        public override BaseLayoutItem CreateLayoutItem(LayoutGroup parent) {
            return new CustomLayoutControlItem((LayoutControlGroup)parent);
        }
        protected override void InitializePaintStyles() {
            ISupportLookAndFeel lookAndFeelOwner = owner.GetISupportLookAndFeel();
            if (lookAndFeelOwner != null) {
                PaintStyles.Add(new LayoutOffice2003PaintStyle(lookAndFeelOwner));
                PaintStyles.Add(new LayoutWindowsXPPaintStyle(lookAndFeelOwner));
                PaintStyles.Add(new CustomLayoutSkinPaintStyle(lookAndFeelOwner));
                PaintStyles.Add(new Style3DPaintStyle(lookAndFeelOwner));
                PaintStyles.Add(new UltraFlatPaintStyle(lookAndFeelOwner));
                PaintStyles.Add(new FlatPaintStyle(lookAndFeelOwner));
            }
            lookAndFeelOwner = null;
        }
    }
}
